public/app/AppWrapper.tsx

```diff
@@ -144,3 +129,124 @@ export class AppWrapper extends Component<AppWrapperProps, AppWrapperState> {
     );
   }
 }
+
+function RouterTree(props: { routes?: JSX.Element | false }) {
+  return (
+    <Router history={locationService.getHistory()}>
+      <LocationServiceProvider service={locationService}>
+        <CompatRouter>
+          <ModalsContextProvider>
+            <GlobalStyles />
+            <div className="grafana-app">
+              <AppChrome>
+                <AngularRoot />
+                <AppNotificationList />
+                <Stack gap={0} grow={1} direction="column">
+                  {pageBanners.map((Banner, index) => (
+                    <Banner key={index.toString()} />
+                  ))}
+                  {props.routes}
+                </Stack>
+                {bodyRenderHooks.map((Hook, index) => (
+                  <Hook key={index.toString()} />
+                ))}
+              </AppChrome>
+            </div>
+            <LiveConnectionWarning />
+            <ModalRoot />
+            <PortalContainer />
+          </ModalsContextProvider>
+        </CompatRouter>
+      </LocationServiceProvider>
+    </Router>
+  );
+}
+
+/**
+ * Renders both the main app tree and a secondary sidecar app tree to show 2 apps at the same time in a resizable split
+ * view.
+ * @param props
+ * @constructor
+ */
+function ExperimentalSplitPaneTree(props: { routes?: JSX.Element | false }) {
+  const appId = useSelector((state) => state.appSidecar.appId);
+  const dispatch = useDispatch();
+
+  const memoryLocationService = new HistoryWrapper(H.createMemoryHistory({ initialEntries: ['/'] }));
+
+  return (
+    <div className="grafana-app">
+      <SplitPaneWrapper
+        splitOrientation="vertical"
+        paneSize={0.25}
+        minSize={200}
+        maxSize={200 * -1}
+        primary="second"
+        splitVisible={!!appId}
+        paneStyle={{ overflow: 'auto', display: 'flex', flexDirection: 'column' }}
+      >
+        {/* very similar to RouterTree but #grafana-app div is moved and it's easier to isolate any changes like this
+         so hopefully less chance of introducing any error */}
```

Comment:
Although maybe it's less chance of introducing an error, but more chance for missing an update that should happen to both of the trees, which could result in discrepancies. 

Couldn't we just move the `.grafana-app` more up the tree for both versions, and with that have a single source of truth?

Reply:
Yeah moved it and does not seem like anything changed. There is some CSS attached:
```
.grafana-app{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:stretch;-webkit-box-align:stretch;-ms-flex-align:stretch;align-items:stretch;position:absolute;width:100%;height:100%;top:0;left:0;}
```

But I think that does not matter that much if it's moved up or down if there isn't any elements rendered in between anyway.